lib/config-array-factory.js

```diff
@@ -405,12 +405,11 @@ function createContext(
  * @returns {Plugin} The normalized plugin.
  */
 function normalizePlugin(plugin) {
-    return {
-        configs: plugin.configs || {},
-        environments: plugin.environments || {},
-        processors: plugin.processors || {},
-        rules: plugin.rules || {}
-    };
+    plugin.configs = plugin.configs || {};
```

Comment:
This can cause an error if `plugin` is frozen or `plugin.configs` is not configurable. I'll send a separate PR.